src/intra_edge.rs

```diff
@@ -173,7 +169,7 @@ impl EdgeBranch {
             EdgeFlags::LEFT_HAS_BOTTOM.union(
                 edge_flags
                     .intersection(EdgeFlags::I420_TOP_HAS_RIGHT)
-                    .select(bl == BL_16X16),
+                    .select(matches!(bl, BlockLevel::Bl16x16)),
```

Comment:
This could still be a `==` comparison right? We're deriving `Eq` for `BlockLevel`. This also applies to the other places where you changed `==` to `matches!`.

Reply:
Okay I suspected that might've been the reason. Thanks for clarifying.

Reply:
One day... https://github.com/joshtriplett/rfcs/blob/is/text/3573-is.md.